examples/schoolyard.jl

```diff
@@ -0,0 +1,138 @@
+# in their [documentation](https://cs.gmu.edu/~eclab/projects/mason/manual.pdf).
+
+# Rather than implementing an Agents.jl&ndash;specific graph structure, we can interface with
+# [LightGraphs.jl](https://github.com/JuliaGraphs/LightGraphs.jl): a high class library for managing
+# and implementing graphs, which can be re-used to establish social networks within existing spaces.
+
+# To begin, we load in some dependencies
+
+using Agents, LightGraphs, SimpleWeightedGraphs, SparseArrays, LinearAlgebra, AgentsPlots
+using Random # hide
+gr() # hide
+
+# And create a very simple agent.
+
+mutable struct Student <: AbstractAgent
+    id::Int
+    pos::Tuple{Float64,Float64}
+end
+
+# ## Rules of the schoolyard
+
+# It's lunchtime, and the students are going out to play.
+# We assume the school building is in the centre of our space, with some fences around the building.
+# A teacher monitors the students, and makes sure they don't stray too far towards the fence.
+# We use a `teacher_attractor` force to simulate a teacher's attentiveness.
+# Students head out to the schoolyard in random directions, but adhere to some social norms.
+
+# Each student has one *friend* and one *foe*. These are chosen at random in our model, so it's
+# possible that for any pair of students, one likes the other but this feeling is not reciprocated.
+# The bond between pairs is chosen at random between 0 and 1, with a bond of 1 being the strongest.
+# If the bond is *friendly*, agents wish above all else to be near their *friend*.
+# Bonds that are *unfriendly* see students moving as far away as possible from their *foe*.
+
+# ## Initialising the model
+
+function schoolyard(;
+    numStudents = 50,
+    teacher_attractor = 0.2,
+    noise = 0.1,
+    max_force = 1.5,
+)
+    model = ABM(
+        Student,
+        ContinuousSpace(2; extend = (100, 100));
+        properties = Dict(
+            :teacher_attractor => teacher_attractor,
+            :noise => noise,
+            :buddies => SimpleWeightedGraph(numStudents),
+            :max_force => max_force,
+        ),
+    )
+    for student in 1:numStudents
+        ## Students begin near the school building
+        add_agent!(model.space.extend .* 0.5 .+ Tuple(rand(2)) .- 0.5, model)
+
+        ## Add one friend and one foe to the social network
+        friend = rand(filter(s -> s != student, 1:numStudents))
+        add_edge!(model.buddies, student, friend, rand())
+        foe = rand(filter(s -> s != student, 1:numStudents))
+        add_edge!(model.buddies, student, foe, -rand())
+    end
+    model
+end
+
+# Our model contains the `buddies` property, which is our LightGraphs.jl undirected, weighted graph.
+# As we can see in the loop, we choose one `friend` and one `foe` at random for each `student` and
+# assign their relationship as a weighted edge on the graph.
+
+# ## Movement dynamics
+
+distance(pos) = sqrt(pos[1]^2 + pos[2]^2)
+scale(L, force) = (L / norm(force)) .* force
```

Comment:
you can replace `norm` with `distance`, and thus not have to use LinearAlgebra at the top.